src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -56,43 +58,99 @@
               optional  = false)
     public String outDirPrefix = null;
 
-    @Argument(doc       = "To run k-mer based read correction, filter and duplication removal in SGA or not, with default parameters.",
+    @Argument(doc       = "To run k-mer based read correction with default parameters (as provided by SGA) or not.",
               shortName = "correct",
-              fullName  = "correctNFilter",
+              fullName  = "kmerCorrectReads",
               optional  = true)
-    public boolean runCorrectionSteps = false;
+    public boolean runCorrection = false;
 
     @Argument(doc       = "If true, capture the stdout and stderr of the SGA processes along with the assembly output (valuable for debugging but hurts performance)",
               shortName = "captureStdIO",
               fullName  = "captureSGAStdOutAndError",
               optional  = true)
     public boolean enableSTDIOCapture = false;
 
+    @Argument(doc       = "An estimate on the number of fastq files in the input directory to be fed to the tool",
+              shortName = "fileCntEst",
+              fullName  = "fastqFileCountEstimate",
+              optional  = true)
+    public int fastqFileCountEstimate = inputSizeNotSet;
+
+
+
+    // for developer performance debugging use
+    private static final DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+    /**
+     * parameter to control how many FASTQ files in each RDD, if {@link RunSGAViaProcessBuilderOnSpark#fastqFileCountEstimate} is not set by user
+     */
+    private static final long NUM_OF_FILES_PER_PARTITION = 10L;
+    private static final int inputSizeNotSet = -1;
+
     @Override
     public void runTool(final JavaSparkContext ctx){
 
+        System.err.println("SGAOnSpark_debug: Start job at " + dateFormat.format(new Date()));
```

Comment:
Please replace this to use a logger:

    private static final Logger logger = LogManager.getLogger(RunSGAViaProcessBuilderOnSpark.class);

...

    logger.log(Level.Debug, ...)